lib/cn/cn_tree_internal.h

```diff
@@ -132,15 +135,17 @@ struct cn_tree {
 
     struct cn_kle_cache ct_kle_cache HSE_L1D_ALIGNED;
 
+    struct mutex      ct_rspills_lock;
+    struct list_head  ct_rspills_list;
+    bool              ct_rspills_wedged;
```

Comment:
Please describe this member as well

Reply:
I can make a halfaxed guess..

Reply:
Oh I thought you just created this.

Reply:
They ensure root spills are committed in order.  Say root spills A, B and C are scheduled in order.  A must commit before B, and B before C.  If C finishes first, it is put on `ct_rspills_list` where it will reside until A and B have been committed.  If A or B fails, then the rspills are wedged and the list must be cleared out and all the operations after the failed one must be canceled.

Reply:
In the new tree, the ordering must be done for each leaf, as opposed to the entire operation.  I think it makes more sense to prevent concurrent root spills from overtaking each other instead of using this rspills list.  We'll get concurrency from different rspills working on different leaves.  Using the previous example, rspill A should be ahead of B.  If B tries to spill to a node that A is still working on, it should just wait so that rspill to any one leaf is serialized.

Reply:
I think for kmt this would effectively serialize all root spills since there are usually keys in every ingest for every node...